app/models/ext_management_system.rb

```diff
@@ -422,6 +422,47 @@ def enable!
     update!(:enabled => true)
   end
 
+  def self.destroy_queue(ids)
+    ids = ids.to_miq_a
+    _log.info("Queuing destroy of #{name} with the following ids: #{ids.inspect}")
+    ids.each do |id|
+      MiqQueue.put(
+        :class_name  => name,
+        :instance_id => id,
+        :method_name => "orchestrate_destroy",
+      )
+    end
+  end
+
+  # override destroy_queue from AsyncDeleteMixin
+  def destroy_queue(deliver_on = nil)
```

Comment:
This method signature is confusing to me.  The instance method takes a time, and the class method takes an array of ids.  In addition, they're very similar with only the deliver_on being the main difference.

Can this method be called `destroy_queue_in_future`, `schedule_destroy_queue` or something else...
and have it call the class method directly, and make the class method take an optional deliver_on along with the ids?